csrc/executor.h

```diff
@@ -179,26 +179,89 @@ class FusionExecutor : public NonCopyable {
     return measure_kernel_time_ ? kernel_time_ms_ : 0;
   }
 
+  //! Returns the input bytes accessed for a kernel
+  //! \note It is important to sample the args struct prior to adding the
+  // 1    output to the args struct
+  int64_t inputBytesProcessed(KernelArgumentHolder& args) {
+    int64_t total_bytes = 0;
+    if (!bytes_processed_per_input_.has_value()) {
+      int64_t num_bytes = 0;
+      bytes_processed_per_input_ = std::optional<std::vector<int64_t>>(
+          std::vector<int64_t>(args.size(), 0));
+      // Figure how many bytes are inputs, outputs, and temporary buffers
+      for (auto i : c10::irange(args.size())) {
+        if (args[i]->is<at::Tensor>()) {
+          auto t = args[i]->as<at::Tensor>();
+          num_bytes = t.numel() *
+              (int64_t)dataTypeSize(aten_to_data_type(t.scalar_type()));
+          bytes_processed_per_input_.value().at(i) = num_bytes;
+          total_bytes += num_bytes;
+        }
+      }
+    } else {
+      for (auto bp : bytes_processed_per_input_.value()) {
```

Comment:
This doesn't look right. Are we going to cache `bytes_processed_per_input_` across `FusionExecutor::runFusion` calls? What if tensor sizes changed?

Reply:
We only store it if the debug dump option is enabled, in which case we retrieve it afterward. So it's always referring to the last call that had the dump option enabled. Note that storing this requires a little bit of computation, so we only want to compute it when we're debugging.

Reply:
The part I didn't get was line 187. It appears to me that we only compute `bytes_processed_per_input_` when it's empty; otherwise the previous computation is reused. Isn't that a problem or am I missing anything?

Reply:
You're right that is a problem.

Reply:
I purposely tried to minimally modify the code that was already there instead of refactoring it.  A lot of this code can likely be removed once we have an alternative to the instrumentation in place.

Reply:
The vector of tensor sizes should go away.  It does not make sense to maintain.  I just don't want to add refactoring away some infrastructure that depends on this vector to this PR.